worker/src/queues/ses/helpers/FileService.ts

```diff
@@ -1,12 +1,23 @@
 import logger, { Logger } from "pino";
 import axios, { AxiosError } from "axios";
 import { ApplicationError } from "../../../utils/ApplicationError";
+import config from "config";
 
 export default class FileService {
   logger: Logger;
+  allowedOrigins: string[];
 
   constructor() {
     this.logger = logger().child({ service: "File" });
+    this.allowedOrigins = config.get<string[]>("Files.allowedOrigins");
+  }
+
+  validateFileLocation(url: string): this {
+    const isAllowed = this.allowedOrigins.find((origin) => url.includes(origin)) !== undefined;
+    if (!isAllowed) {
+      throw new ApplicationError("FILE", "ORIGIN_NOT_ALLOWED", `The specified file location ${url} is forbidden`);
+    }
+    return this;
```

Comment:
if url is something like 
`http://not-ok.com?param=http://notarial-api` - this will end up passing the check.

```
const url = "http://not-ok.com?param=http://notarial-api:9000"
const allowedOrigins = ["http://notarial-api"]
const isAllowed = allowedOrigins.find((origin) => url.includes(origin)) !== undefined;
console.log(isAllowed) // true
```

in the past I've used the URL class to both extract origin, hostname etc, and do a match/find/etc against that.

```
 URL {
  href: 'http://not-ok.com/?param=http://notarial-api:9000',
  origin: 'http://not-ok.com',
  protocol: 'http:',
  username: '',
  password: '',
  host: 'not-ok.com',
  hostname: 'not-ok.com',
  port: '',
  pathname: '/',
  search: '?param=http://notarial-api',
  searchParams: URLSearchParams { 'param' => 'http://notarial-api:9000' },
  hash: ''
}
```

I'd probably wrap the whole thing in a try/catch and just emit the same error for invalid URL or not allowed origin.